qmi/instruments/high_finesse/wlm.py

```diff
@@ -124,35 +186,130 @@ def get_frequency(self, channel: int) -> float:
     def get_wavelength(self, channel: int) -> float:
         """Get the main results of the measurement of a specified signal.
 
-        :param channel: The signal number (1 to 8) in case of a WLM with multichannel switch or with double pulse
-            option (MLC). For WLMs without these options 1 should be overhanded.
-        :return: The last measured wavelength in nm.
-        :raises QMI_InstrumentException: In case of instrument error or channel out of range.
+        Parameters:
+            channel: The signal number (1 to 8) in case of a WLM with multichannel switch or with double pulse
+                     option (MLC). For WLMs without these options 1 should be overhanded.
+
+        Returns:
+            wavelength: The last measured wavelength in nm.
         """
-        _logger.info("[%s] Getting wavelength on channel %d", self._name, channel)
+        _logger.debug("[%s] Getting wavelength on channel %d", self._name, channel)
         self._check_channel(channel)
 
         wavelength = self._lib.dll.GetWavelengthNum(channel, 0.0)
 
         return self._check_for_error_code(wavelength, "GetWavelengthNum")
 
     @rpc_method
-    def close(self) -> None:
-        """Close the connection to the instrument hardware and release associated resources."""
-        _logger.info("[%s] Closing connection to instrument", self._name)
+    def get_power(self, channel: int) -> float:
+        """Get the power of the current measurement shot of a specified signal.
 
-        super().close()
+        Parameters:
+            channel: The signal number (1 to 8) in case of a WLM with multichannel switch or with double pulse
+                     option (MLC). For WLMs without these options 1 should be overhanded.
 
-    def _check_channel(self, channel: int) -> None:
-        if not (0 < channel <= self.MAX_CHANNEL_NUMBER):
-            raise QMI_InstrumentException(f"Channel number out of range: {channel}")
+        Returns:
+            power:   The power of the last measured cw or quasi cw signal in uW or the energy in uJ.
+        """
+        _logger.debug("[%s] Getting power on channel %d", self._name, channel)
+        self._check_channel(channel)
 
-    def _check_for_error_code(self, value: float, method: str) -> float:
-        if value > 0:
-            return value
+        power = self._lib.dll.GetPowerNum(channel, 0.0)
 
-        err_msg = WS8_ERR(value) if value in [error.value for error in WS8_ERR] else str(value)
+        return self._check_for_error_code(power, "GetPowerNum")
 
-        _logger.warning("[%s] %s error: %s", self._name, method, err_msg)
+    @rpc_method
+    def read_measurement(self, channel: int) -> Measurement:
```

Comment:
My preference would be to do these type of abstractions outside the drivers. For example, the wavelength monitoring service, we can have a generic/abstract container that we can pipe data from bristol/wlm into. So I think the easiest is just to remove this method and the import to bristol.